internal/artifacts/module_arch_test.go

```diff
@@ -703,8 +792,114 @@ module-types:
 		})
 	})
 
+	Describe("checkBuildResultsConflicts", func() {
+		DescribeTable("conflicting and none conflicting cases", func(target string, modules []string, result func() types.GomegaMatcher) {
+			mtaObj := getMtaObj("mtahtml5", "mta.yaml")
+			Ω(checkBuildResultsConflicts(mtaObj, getTestPath("mtahtml5"), target, nil, modules, os.Getwd)).Should(result())
+		},
+			Entry("one module, no conflicts", getTestPath("result"), []string{"ui5app"}, Succeed),
+			Entry("2 none conflicting modules, because target not provided and modules results are in different sub folders", "", []string{"ui5app", "ui5app2"}, Succeed),
+			Entry("2 conflicting modules, because they share the same provided target", getTestPath("result"), []string{"ui5app", "ui5app2"}, HaveOccurred),
+		)
+
+		It("fails on location initialization", func() {
+			mtaObj := getMtaObj("mtahtml5", "mta.yaml")
+			Ω(checkBuildResultsConflicts(mtaObj, getTestPath("mtahtml5"), "", nil, []string{"ui5app"}, failingGetWd)).Should(HaveOccurred())
+		})
+
+		DescribeTable("fails on the mta yaml problems", func(filename string) {
+			mtaObj := getMtaObj("mtahtml5", filename)
+			Ω(checkBuildResultsConflicts(mtaObj, getTestPath("mtahtml5"), getTestPath("result"), nil, []string{"ui5app"}, os.Getwd)).Should(HaveOccurred())
+		},
+			Entry("unknown builder", "mtaWithNoCustomCommands.yaml"),
+			Entry("wrong definition of the build result property", "mtaWithWrongBuildResult.yaml"))
+	})
+
+	Describe("getModuleLocation", func() {
+		It("sanity, target provided", func() {
+			loc, err := getModuleLocation(getTestPath(), getTestPath("result"), "m1", nil, os.Getwd)
+			Ω(err).Should(Succeed())
+			Ω(loc.GetTargetModuleDir("m1")).Should(Equal(getTestPath("result")))
+		})
+
+		It("sanity, target not provided - default one is used", func() {
+			loc, err := getModuleLocation(getTestPath("mta"), "", "m1", nil, os.Getwd)
+			Ω(err).Should(Succeed())
+			currentFolder, err := os.Getwd()
+			Ω(err).Should(Succeed())
+			Ω(loc.GetTargetModuleDir("m1")).Should(Equal(filepath.Join(currentFolder, ".mta_mta_build_tmp", "m1")))
+		})
+
+		It("fails on default target detecting", func() {
+			_, err := getModuleLocation("", "", "m1", nil, failingGetWd)
+			Ω(err).Should(HaveOccurred())
+		})
+
+		// this test was added for coverage purpose only,
+		// in case of problems with getWd function failure will happen before location initialization
+		It("fails on location initialization", func() {
+			count := 0
+			_, err := getModuleLocation("", "", "m1", nil, func() (string, error) {
+				if count == 0 {
+					count++
+					return "", nil
+				}
+				return "", errors.New("error")
+			})
+
+			Ω(err).Should(HaveOccurred())
+		})
+	})
+
+	Describe("getRequiredModules", func() {
+
+		var mtaObj *mta.MTA
+
+		BeforeEach(func() {
+			mtaObj = getMtaObj("mtahtml5", "mtaWithWrongBuildRequirements.yaml")
+		})
+
+		Describe("scenarios based on mtaWithWrongBuildRequirements.yaml", func() {
```

Comment:
Why do you need this describe? Can't the tests be directly under the parent describe?

Reply:
removed